aea/cli/check_packages.py

```diff
@@ -324,22 +347,261 @@ def check_public_id(configuration_file: Path) -> None:
+            location = packages_info[0]["location"]
+        else:
+            files = packages_info[0].files  # type: ignore
+            location = packages_info[0].location  # type: ignore
+        return [Path(location) / i for i in files]  # type: ignore
+
+    @staticmethod
+    def clean_dependency_name(dependecy_specification: str) -> str:
+        """Get dependency name from dependency specification."""
+        match = DEP_NAME_RE.match(dependecy_specification)
+        if not match:
+            raise ValueError(f"Bad dependency specification: {dependecy_specification}")
+        return match.groups()[0]
+
+
+class ImportsTool:
+    """Tool to work with 3rd part imports in source code."""
+
+    @staticmethod
+    def get_imports_for_file(pyfile: Union[str, Path]) -> List[str]:
+        """Get all imported modules for python source file."""
+        with open(pyfile, "r") as f:
+            statements = f.read()
+        instructions = dis.get_instructions(statements)  # type: ignore
+        imports = []
+        is_try_except_block = False
+        for im in instructions:
+            if "SETUP_FINALLY" in im.opname:
+                is_try_except_block = True
+
+            if "POP_EXCEPT" in im.opname:
+                is_try_except_block = False
+
+            if is_try_except_block:
+                continue
+
+            if "IMPORT" in im.opname:
+                imports.append(im)
+
+        grouped: Dict[str, List[str]] = defaultdict(list)
+        for instr in imports:
+            grouped[instr.opname].append(instr.argval)
+
+        return grouped["IMPORT_NAME"]
+
+    @staticmethod
+    def get_module_file(module_name: str) -> str:
+        """Get module source file name."""
+        try:
+            mod = importlib.import_module(module_name)
+            return getattr(mod, "__file__", "")
+        except (AttributeError, ModuleNotFoundError):
+            return ""
+
+    @staticmethod
+    @list_decorator
+    def list_all_pyfiles(
+        root_path: Union[Path, str],
+        pattern: str = "**/*.py",
+        ignores: Optional[Set[Path]] = None,
+    ) -> Generator:
+        """List all python files in directory."""
+        ignores = ignores or set()
+        root_path = Path(root_path)
+        for path in root_path.glob(pattern):
+            # check if path is a subpath of an ignore path
+            relative_path = path.relative_to(root_path)
+            if any(os.path.commonprefix([relative_path, p]) != "" for p in ignores):
+                continue
+            yield path.relative_to(root_path)
+
+    @classmethod
+    @list_decorator
+    def get_third_part_imports_for_file(cls, pyfile: str) -> Generator:
+        """Get list of third part modules imported for source file."""
+        imports = cls.get_imports_for_file(pyfile)
+        for module_name in imports:
+            pyfile = cls.get_module_file(module_name)
+            if not pyfile:
+                continue
```

Comment:
this helper function / static method filters out the `builtins` and other C-level modules?